builtin/providers/google/resource_compute_backend_service.go

```diff
@@ -0,0 +1,410 @@
+package google
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/googleapi"
+)
+
+func resourceComputeBackendService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceComputeBackendServiceCreate,
+		Read:   resourceComputeBackendServiceRead,
+		Update: resourceComputeBackendServiceUpdate,
+		Delete: resourceComputeBackendServiceDelete,
+
+		Schema: map[string]*schema.Schema{
+			"backend": &schema.Schema{
+				Type: schema.TypeSet,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"balancing_mode": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "UTILIZATION",
+						},
+						"capacity_scaler": &schema.Schema{
+							Type:     schema.TypeFloat,
+							Optional: true,
+							Default:  1,
+						},
+						"description": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"group": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"max_rate": &schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"max_rate_per_instance": &schema.Schema{
+							Type:     schema.TypeFloat,
+							Optional: true,
+						},
+						"max_utilization": &schema.Schema{
+							Type:     schema.TypeFloat,
+							Optional: true,
+							Default:  0.8,
+						},
+					},
+				},
+				Optional: true,
+				Set:      resourceGoogleComputeBackendServiceBackendHash,
+			},
+
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"health_checks": &schema.Schema{
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Required: true,
+				Set:      schema.HashString,
+			},
+
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
```

Comment:
Yeah, it'd be useful for all resources

On Sat, Aug 22, 2015, 5:11 PM Radek Simko <notifications@github.com> wrote:

> In builtin/providers/google/resource_compute_backend_service.go
> <https://github.com/hashicorp/terraform/pull/3038#discussion_r37699801>:
>
> > +				Type:     schema.TypeString,
> > +				Optional: true,
> > +			},
> > +
> > +			"health_checks": &schema.Schema{
> > +				Type:     schema.TypeSet,
> > +				Elem:     &schema.Schema{Type: schema.TypeString},
> > +				Required: true,
> > +				Set:      schema.HashString,
> > +			},
> > +
> > +			"name": &schema.Schema{
> > +				Type:     schema.TypeString,
> > +				Required: true,
> > +				ForceNew: true,
> > +				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
>
> Factor out what exactly? That validation function you mean?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hashicorp/terraform/pull/3038/files#r37699801>.
>